### DISCUSSION TITLE
Why there clone repository with submodule were empty?

### DISCUSSION THREAD

**Discussion by elkan1788** [vercel/vercel#8683] [Help]
Hi guys, I had try used Vercel support [deploy button](https://vercel.com/docs/deploy-button) to make my repository as template , that could be help people who wan't deploy their blog site on Vercel. but not good and  meet something wrong. 

My repository is here:  [hugo-theme-next-starter](https://github.com/hugo-next/hugo-theme-next-starter)
My Custom deploy button is :  [![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/clone?repository-url=https%3A%2F%2Fgithub.com%2Fhugo-next%2Fhugo-theme-next-starter&env=HUGO_VERSION&envDescription=Enter%20latest%20version%20of%20Hugo%20engine.&envLink=https%3A%2F%2Fgithub.com%2Fgohugoio%2Fhugo%2Freleases%2F&project-name=my-blog&repo-name=my-blog&demo-title=Hugo%20NexT%20Theme&demo-description=Easily%20%26%20powerful%20theme%20of%20Hugo%20engine.&demo-url=https%3A%2F%2Fhugo-next.eu.org%2F&demo-image=https%3A%2F%2Fimgs.lisenhui.cn%2Fhugo-next%2Fhugo-next-demo.png)

After I used Vercel deploy button start new project on Vercel,  there found it created new repository in Github was well, but I found that it don't clone the submodule status correctly, the submodule folder is empty, so when Vercel deploy were error because there not clone submodule folder things.

There seems only could used [Git provider](https://vercel.com/docs/concepts/git) by manual to create new project as well, So how colud I do to fixed this issue?

Thanks.

**Comment by amyegan** [comment]
Hi @elkan1788! By default, you get the submodule directories without files in them when you [clone a project with submodules](https://www.git-scm.com/book/en/v2/Git-Tools-Submodules#_cloning_submodules). You would need to use `--recurse-submodules` on the clone command. I'm not sure if the deploy button enables that customization, but I'm asking the team about it and will let you know when I learn more.

**Reply by elkan1788** [reply]
@amyegan Thanks , I know what your suggest, but there need Vercel `deploy-button` could enable custom comand. I had tried to added `git submodule update --init --recursive` command into `install commnad` parameter as value that not good and it could not work well. It's there have any other way to solution this issue?

**Reply by amyegan** [reply]
Sorry for the delay. I have asked a few different teams and found that we do not have a simple way to handle the submodule cloning 😞 

I don't have a lot of experience working with Hugo and submodules, so I might be missing something obvious. I'm seeing some behavior differences between a repository cloned the GitHub "Use this template" button and one cloned from the Vercel "Deploy" button. For the repo cloned by the deploy button, I needed to run a different command from my local computer to add the submodule because the subproject commit information was missing (source: https://stackoverflow.com/a/9049703)

I'm adding "submodule cloning suport" to the list of requested features. But, for now, it does not seem like you'll be able to get a good user experience for a project using submodules with the deploy button.

**Reply by elkan1788** [reply]
OK, thanks  for your supported. I will wating this feature release and hope it come in true  as soom as possible. :smile:

**Comment by elkan1788** [comment]
Is there other people have any workaround to resolve this issue?

Thanks.
